libs/repositories/src/UsdRepository/UsdRepositoryCoingecko.ts

```diff
@@ -0,0 +1,114 @@
+import 'reflect-metadata';
+
+import { injectable } from 'inversify';
+import { PricePoint, PriceStrategy, UsdRepository } from './UsdRepository';
+import { COINGECKO_PLATFORMS, coingeckoProClient } from '../coingecko';
+
+/**
+ * Number of days of data to fetch for each price strategy
+ *
+ * Depending on the granularity, we specify the number of days of data to fetch.
+ * Coingecko will auto-choose the granularity based on the number of days. Its actually weird, because if you try to
+ * specify the granularity it will throw you an error (saying that the PRO account is not enough). So its important to
+ * control the granularity by the number of days.
+ *
+ */
+const DAYS_PER_PRICE_STRATEGY: Record<PriceStrategy, number> = {
+  '5m': 1, // 1 day (~288 points)
+  hourly: 5, // 5 Days of hourly data (~120 points)
+  daily: 90, // 90 Days of daily data (~90 points)
+};
+
+@injectable()
+export class UsdRepositoryCoingecko implements UsdRepository {
+  async getUsdPrice(
+    chainId: number,
+    tokenAddress: string
+  ): Promise<number | null> {
+    const platform = COINGECKO_PLATFORMS[chainId];
+    if (!platform) {
+      return null;
+    }
+
+    const tokenAddressLower = tokenAddress.toLowerCase();
+
+    // Get prices: See https://docs.coingecko.com/reference/contract-address-market-chart
+    // Get prices. See https://docs.coingecko.com/reference/coins-id-market-chart
+    const fetchResponse = await coingeckoProClient.GET(
+      `/simple/token_price/{id}`,
+      {
+        params: {
+          path: {
+            id: platform,
+          },
+          query: {
+            contract_addresses: tokenAddressLower,
+            vs_currencies: 'usd',
+          },
+        },
+      }
+    );
+
+    if (fetchResponse.error) {
+      throw fetchResponse.error;
+    }
+    const priceData = fetchResponse.data;
+
+    if (!priceData[tokenAddressLower] || !priceData[tokenAddressLower].usd) {
+      return null;
+    }
+
+    return priceData[tokenAddressLower].usd;
+  }
+
+  async getUsdPrices(
+    chainId: number,
+    tokenAddress: string,
+    priceStrategy: PriceStrategy
+  ): Promise<PricePoint[] | null> {
+    const platform = COINGECKO_PLATFORMS[chainId];
+    if (!platform) {
+      return null;
+    }
+
+    const tokenAddressLower = tokenAddress.toLowerCase();
+    const days = DAYS_PER_PRICE_STRATEGY[priceStrategy].toString();
```

Comment:
Not what I meant 😅 

Store them in the object as string instead of number, so the casting here is not needed.
This is not used anywhere else anyway.

Reply:
i could, but it is a number. So yours would be a performance nit (also to make it simpler), but i think is nicer to keep it as the raw type, no? After all is weird to work with a number of days as string

Reply:
Ok, not strongly against it.
I just like to not have casting when not needed.